### PR TITLE
Feat/deeplinking

### PR DESCRIPTION
**Pull request by AndreasGassmann**

### CODE REVIEW

File: src/clients/beacon-client/BeaconClient.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export abstract class BeaconClient {
     this.initSDK().catch(console.error)
   }
 
-  public async resetSDK(): Promise<void> {
+  public async destroy(): Promise<void> {
```
Comment (godenzim) [inline src/clients/beacon-client/BeaconClient.ts:53]: @AndreasGassmann please add the documentation for this method explaining when to use it and the consequences (client instance cannot be used anymore).

File: src/events.ts [hunk 2]
```diff
@@ -123,14 +124,20 @@ const showNoPermissionAlert = async (): Promise<void> => {
...
-const showErrorAlert = async (beaconError: ErrorResponse): Promise<void> => {
+const showErrorAlert = async (
+  beaconError: ErrorResponse,
+  buttons?: AlertButton[]
+): Promise<void> => {
   const error = beaconError.errorType
     ? BeaconError.getError(beaconError.errorType)
     : new UnknownBeaconError()
 
+  console.log('showing error alert type ', beaconError.errorType)
```
Comment (godenzim) [inline src/events.ts:135]: @AndreasGassmann do we really need this `console.log`?

File: src/events.ts [hunk 3]
```diff
@@ -192,24 +198,20 @@ const showQrCode = async (
   console.log(dataString)
 
   const base58encoded = await serializer.serialize(data)
-  console.log('base58encoded pair code', base58encoded)
```
Comment (godenzim) [inline src/events.ts:195]: @AndreasGassmann and this ones?

File: src/clients/dapp-client/DAppClient.ts [hunk 4]
```diff
@@ -529,8 +530,44 @@ export class DAppClient extends Client {
...
+
+          // if (!activeAccount) {
+          //   // send new permission request
+          //   await this.requestPermissions({
+          //     network: accountInfo?.network,
+          //     scopes: accountInfo?.scopes
+          //   })
+          // }
+          // // send operation again
+          // await this.requestOperation({ operationDetails: operationRequest.operationDetails })
```
Comment (godenzim) [inline src/clients/dapp-client/DAppClient.ts:563]: @AndreasGassmann if this code is not needed anymore just remove it

File: src/events.ts [hunk 5]
```diff
@@ -192,24 +198,20 @@ const showQrCode = async (
...
+  console.log(base58encoded)
```
Comment (godenzim) [inline src/events.ts:201]: @AndreasGassmann and this ones?

Comment (AndreasGassmann) [reply src/events.ts]: Yes, we need this until we have a "copy to clipboard" feature. I added a comment.

**Review by godenzim — APPROVED** [review]

**Pull request merged**